12-target-details.tex

```diff
@@ -140,10 +140,71 @@ \subsection{Javascript target Metatags}
 	\hline
 	\multicolumn{3}{|c|}{Javascript metatags} \\ \hline
 	Metatag &  Description & Target \\ \hline
-	@:expose \_(?Name=Class path)\_  &  Makes the class available on the \expr{window} object  & js  \\
+	@:expose \_(?Name=Class path)\_  &  Makes the class available on the \expr{window} object or \expr{exports} for node.js  & js  \\
 \end{tabular}
 \end{center}
 
+\subsection{Exposing Haxe classes for Javascript}
+\label{target-javascript-expose}
+
+It is possible to make Haxe classes or static fields available for usage in plain Javascript. 
+To expose, apply the \expr{@:expose} metadata on the desired class or static fields.
+
+This example exposes the Haxe class \ic{MyClass}.
+
+\haxe{assets/ClassExpose.hx}
+
+This generates the following Javascript output:
+
+\begin{lstlisting}
+(function ($hx_exports) { "use strict";
+var MyClass = $hx_exports.MyClass = function(name) {
+	this.name = name;
+};
+MyClass.prototype = {
+	foo: function() {
+		return "Greetings from " + this.name + "!";
+	}
+};
+})(typeof window != "undefined" ? window : exports);
+\end{lstlisting}
+
+By passing globals (like \ic{window} or \ic{exports}) as parameters to our anonymous function in the Javascript module, it is available which allows to expose the Haxe generated module.
+
+In plain Javascript it is now possible to create an instance of the class and call its public functions.
+
+\begin{lstlisting}
+// Javascript code
+var instance = new MyClass('Mark');
+console.log(instance.foo()); // logs a message in the console
+\end{lstlisting}
+
+The package path of the Haxe class is in the same way fully exposed. To rename the class or define a different package for the exposed class use \expr{@:expose("my.package.MyExternalClass")}
+
+\paragraph{Shallow expose}
+
+When the Haxe generated code is part of a larger Javascript project and wrapped in a large closure it is not always necessary to expose the Haxe types to global variables.
+Compiling the project using \ic{-D shallow-expose} permits the types or static fiels to be available for the surrounding scope of the Haxe generated closure only.
```

Comment:
"allows" instead of "permits" also, typo in "fields" and "Haxe-generated" @Simn maybe we can agree to not capitalize Haxe in this instance, beacause it is used as an adjective and not as a name.